tests/validator/validator.cpp

```diff
@@ -846,6 +846,136 @@ TEST(Validator, validMathMLCiAndCnElementsWithCellMLUnits)
     EXPECT_EQ(size_t(0), v->issueCount());
 }
 
+TEST(Validator, invalidMathMLDegreeOutsideRootOrDiff)
+{
+    const std::string e = "Math has a 'degree' element in a parent 'apply' element; 'degree' is supported for 'root' or 'diff' elements only.";
+    const std::string math =
+        "<math xmlns=\"http://www.w3.org/1998/Math/MathML\">\n"
+        "  <apply>\n"
+        "    <eq/>\n"
+        "    <ci>C</ci>\n"
+        "    <apply>\n"
+        "      <degree/>\n"
+        "      <ci>A</ci>\n"
+        "      <ci>B</ci>\n"
+        "    </apply>\n"
+        "  </apply>\n"
+        "</math>\n";
+
+    libcellml::ValidatorPtr v = libcellml::Validator::create();
+    libcellml::ModelPtr m = libcellml::Model::create();
+    libcellml::ComponentPtr c = libcellml::Component::create();
+    libcellml::VariablePtr v1 = libcellml::Variable::create();
+    libcellml::VariablePtr v2 = libcellml::Variable::create();
+    libcellml::VariablePtr v3 = libcellml::Variable::create();
+
+    m->setName("modelName");
+    c->setName("componentName");
+    v1->setName("A");
+    v2->setName("B");
+    v3->setName("C");
+    v1->setInitialValue("1.0");
+    v2->setInitialValue("-1.0");
+    v1->setUnits("dimensionless");
+    v2->setUnits("dimensionless");
+    v3->setUnits("dimensionless");
+
+    c->addVariable(v1);
+    c->addVariable(v2);
+    c->addVariable(v3);
+    c->setMath(math);
+    m->addComponent(c);
+
+    v->validateModel(m);
+    EXPECT_EQ(size_t(1), v->errorCount());
+    EXPECT_EQ(e, v->error(0)->description());
```

Comment:
I would use `EXPECT_EQ_ISSUES()`.

Reply:
I just copied from the code above ... you know, for consistency ;)